src/components/Comment/CommentSection/CommentSection.tsx

```diff
@@ -59,11 +49,68 @@ export default function CommentSection({
       await createComment(eventId, commentBody, replyToCommentId);
       refreshEvent();
       setCommentBody('');
-      setReplyToCommentAuthor(undefined);
-      setReplyToCommentId(undefined);
+      handleAbortReply();
     }
   };
 
+  const setTextAreaFocus = () => {
+    textAreaRef.current?.focus();
+  };
+
+  const handleAbortReply = () => {
+    setReplyToCommentId(undefined);
+    setReplyToCommentAuthor(undefined);
+  };
+
+  useEffect(() => {
+    const topLevelComments = comments?.filter(
+      (comment) => !comment.replyToCommentId
+    );
+
+    const handleClickReply = (commentData: CommentData) => {
+      setReplyToCommentAuthor(commentData.byPerson.firstName);
+      setReplyToCommentId(commentData.id);
+      setTextAreaFocus();
+    };
+
+    const generateCommentNodes = (
+      comments: Array<CommentData> | undefined,
+      handleClickReply: (commentData: CommentData) => void,
+      isTopLevel = true
+    ): ReactNode => {
+      if (!comments) {
+        return undefined;
+      }
+
+      return (
+        <CommentGroup>
+          {comments.map((comment) => {
+            let commentChildrenNodes = generateCommentNodes(
+              comment.children,
+              handleClickReply,
+              false
+            );
+
+            return (
+              <Comment
+                key={comment.id}
+                commentData={comment}
+                isTopLevel={isTopLevel}
+                onReply={handleClickReply}
+              >
+                {commentChildrenNodes}
+              </Comment>
+            );
+          })}
+        </CommentGroup>
+      );
+    };
```

Comment:
It didn't work with a recursively defined component?

Moving this function into the component actually makes the readbility worse 😅. It's better to move as much logic outside of the component as possible